### ISSUE TITLE
provide ways to override browser in WithBrowser abstract class

### ISSUE THREAD

**Issue by antonkulaga** [playframework/playframework#3777]
Providing browser that  org.fluentlenium.core.FluentAdapter is not convenient for those who use other selenium wrappers (like https://github.com/johanandren/scalenium ). I suggest to provide a withbrowser class that I will be able to inherit and define my own browser implementation.

**Comment by julienrf** [comment]
Hi, you can override the `provideBrowser` method. https://www.playframework.com/documentation/2.3.x/api/java/index.html

**Comment by antonkulaga** [comment]
What should I override in scala api?

**Comment by julienrf** [comment]
You can override the `browser` member or just pass a selenium driver as parameter.
https://www.playframework.com/documentation/2.3.x/api/scala/index.html#play.api.test.WithBrowser

**Comment by julienrf** [comment]
You are totally right. Maybe an extension point could be added to handle your use case, or maybe you could write your own helper. The implementation quite straightforward: https://github.com/playframework/playframework/blob/2.3.x/framework/src/play-test/src/main/scala/play/api/test/Specs.scala#L48-L70
